crates/ruma-client-api/src/media.rs

```diff
@@ -1,8 +1,12 @@
 //! Endpoints for the media repository.
 
 pub mod create_content;
+#[cfg(feature = "unstable-msc2246")]
+pub mod create_mxc_uri;
 pub mod get_content;
 pub mod get_content_as_filename;
 pub mod get_content_thumbnail;
 pub mod get_media_config;
 pub mod get_media_preview;
+#[cfg(feature = "unstable-msc2246")]
+pub mod upload_async_media;
```

Comment:
Since this is just a variation of `create_content` IIUC, I would prefer a name closer to that, maybe `create_content_async`.